moveit_plugins/moveit_fake_controller_manager/src/moveit_fake_controller_manager.cpp

```diff
@@ -35,152 +35,156 @@
-    if (node_handle_.getParam("initial", initial))
+    if (node_->has_parameter(PARAM_BASE_NAME + ".initial"))
     {
-      sensor_msgs::JointState js = loadInitialJointValues(initial);
-      js.header.stamp = ros::Time::now();
-      pub_.publish(js);
+      sensor_msgs::msg::JointState js = loadInitialJointValues(PARAM_BASE_NAME + ".initial");
+      js.header.stamp = rclcpp::Clock(RCL_ROS_TIME).now();
+      pub_->publish(js);
     }
 
+    std::vector<std::string> controller_names = controller_names_param.as_string_array();
     /* actually create each controller */
-    for (int i = 0; i < controller_list.size(); ++i)  // NOLINT(modernize-loop-convert)
+    for (const std::string& controller_name : controller_names)
     {
-      if (!controller_list[i].hasMember("name") || !controller_list[i].hasMember("joints"))
-      {
-        ROS_ERROR_NAMED("MoveItFakeControllerManager", "Name and joints must be specified for each controller");
-        continue;
-      }
-
       try
       {
-        const std::string name = std::string(controller_list[i]["name"]);
-
-        if (controller_list[i]["joints"].getType() != XmlRpc::XmlRpcValue::TypeArray)
+        std::vector<std::string> controller_joints;
+        if (!node_->get_parameter(PARAM_BASE_NAME + "." + controller_name + ".joints", controller_joints))
         {
-          ROS_ERROR_STREAM_NAMED("MoveItFakeControllerManager", "The list of joints for controller "
-                                                                    << name << " is not specified as an array");
+          RCLCPP_ERROR_STREAM(LOGGER, "No joints specified for controller " << controller_name);
           continue;
         }
-        std::vector<std::string> joints;
-        joints.reserve(controller_list[i]["joints"].size());
-        for (int j = 0; j < controller_list[i]["joints"].size(); ++j)
-          joints.emplace_back(std::string(controller_list[i]["joints"][j]));
 
-        const std::string& type =
-            controller_list[i].hasMember("type") ? std::string(controller_list[i]["type"]) : DEFAULT_TYPE;
+        std::string type = DEFAULT_TYPE;
+        if (node_->has_parameter(PARAM_BASE_NAME + "." + controller_name + ".type"))
+          node_->get_parameter(PARAM_BASE_NAME + "." + controller_name + ".type", type);
+
         if (type == "last point")
-          controllers_[name].reset(new LastPointController(name, joints, pub_));
+          controllers_[controller_name].reset(new LastPointController(controller_name, controller_joints, pub_));
         else if (type == "via points")
-          controllers_[name].reset(new ViaPointController(name, joints, pub_));
+          controllers_[controller_name].reset(new ViaPointController(controller_name, controller_joints, pub_));
         else if (type == "interpolate")
-          controllers_[name].reset(new InterpolatingController(name, joints, pub_));
+        {
+          // @note: would put this in InterpolatingController's constructor
```

Comment:
Why is this required now? If this is really only used in the `InterpolatingController` I would agree that we should move it there.

Reply:
remove comment 0f02f39